Makefile

```diff
@@ -170,15 +172,18 @@ cluster/cleanup/crds:
 	@-oc delete crd installations.integreatly.org
 	@-oc delete crd webapps.integreatly.org
 
+.PHONY: deploy/integreatly-installation-cr.yml
 deploy/integreatly-installation-cr.yml: export SELF_SIGNED_CERTS := true
-deploy/integreatly-installation-cr.yml: export INSTALLATION_NAME := example-installation
+deploy/integreatly-installation-cr.yml: export INSTALLATION_NAME := integreatly-operator
 deploy/integreatly-installation-cr.yml: export INSTALLATION_TYPE := managed
+deploy/integreatly-installation-cr.yml: export INSTALLATION_PREFIX := rhmi
 deploy/integreatly-installation-cr.yml:
 	@echo "selfSignedCerts = $(SELF_SIGNED_CERTS)"
-	sed "s/INSTALLATION_NAME/$(INSTALLATION_NAME)/g" | \
+	sed "s/INSTALLATION_NAME/$(INSTALLATION_NAME)/g" deploy/crds/examples/integreatly-installation-cr.yaml | \
 	sed "s/INSTALLATION_TYPE/$(INSTALLATION_TYPE)/g" | \
+	sed "s/INSTALLATION_PREFIX/$(INSTALLATION_PREFIX)/g" | \
 	sed "s/SELF_SIGNED_CERTS/$(SELF_SIGNED_CERTS)/g" > deploy/integreatly-installation-cr.yml
-
+	@-oc create -f deploy/integreatly-installation-cr.yml
```

Comment:
Ah, I guess this is why it's a phony target now.

As I see it, there are a couple of things here:
- It creates the file and then runs `oc create`, so even if the file exists, it's a phony target to make sure `oc create` is run even if the file already existed

BUT!

- If the file already exists, then we have to assume that this was already run, so `oc create` was already run; and running `oc create` twice does not bring joy (even if we are ignoring failures)

There are a couple of ideas that could be done to improve on this I think (pick and choose if you think they're good):

- We could move the `oc create` away, and make it a boring ol' non-phony file target again
- If the file isn't actually important, we could remove creation of it, and just pipe to `oc create -f -`
- _Maybe_ we should default to `oc apply` rather than `oc create` in make recipes (not 100% sure on that, just a feeling)

Reply:
Reason why I did this was now both `test/e2e` and `test/e2e/olm` depends on `deploy/integreatly-installation-cr.yml` and needed to create the cr also

`cluster/deploy/integreatly-installation-cr.yml` depends on `deploy/integreatly-installation-cr.yml` beforehand anyhow,  previously with the `oc create` command to create the cr so just wanted to remove the extra need of havin the same `oc create` in the `test/e2e`

> We could move the oc create away, and make it a boring ol' non-phony file target again


Yeah, I'm good with moving the `oc create` away again but I would need to add it back in to `cluster/deploy/integreatly-installation-cr.yml` and also to `test/e2e`

> If the file isn't actually important, we could remove creation of it, and just pipe to oc create -f - 


I think the file is important since it looks like its used in `cluster/cleanup/olm`

> Maybe we should default to oc apply rather than oc create in make recipes (not 100% sure on that, just a feeling)


Not sure on this, haven't used `oc apply` much but I think it should work